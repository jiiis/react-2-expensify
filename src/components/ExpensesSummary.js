import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'

import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => (
  <div>Viewing {expenseCount} expense{expenseCount > 1 ? 's' : ''} totalling {numeral(expensesTotal / 100).format('$0,0.00')}</div>
)

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  }
}

export default connect(mapStateToProps)(ExpensesSummary)
