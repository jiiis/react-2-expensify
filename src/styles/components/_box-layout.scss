.box-layout {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  background: url('/images/bg.jpg');
  background-size: cover;
}

.box-layout__box {
  padding: $l-size $m-size;
  width: 25rem;
  background: fade-out(#fff, .15);
  border-radius: 3px;
  text-align: center;
}

.box-layout__title {
  margin: 0 0 $m-size;
  line-height: 1;
}
