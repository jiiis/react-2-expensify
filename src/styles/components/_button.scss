.button {
  display: inline-block;
  padding: $s-size;
  line-height: 1;
  border: none;
  font-size: $fond-size-large;
  font-weight: 300;
  text-decoration: none;
  color: #fff;
  background: $blue;
}

.button--secondary {
  background: #888;
}

.button--link {
  background: none;
}
