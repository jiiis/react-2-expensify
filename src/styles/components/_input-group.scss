.input-group {
  display: flex;
  flex-direction: column;
  margin: 0 0 $m-size;

  @media (min-width: $desktop-breakpoint) {
    flex-direction: row;
    margin: 0 0 $l-size;
  }
}

.input-group__item {
  margin: 0 0 $s-size 0;

  @media (min-width: $desktop-breakpoint) {
    margin: 0 $s-size 0 0;
  }
}
