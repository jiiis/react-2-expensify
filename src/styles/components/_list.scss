.list-header {
  display: flex;
  justify-content: space-between;
  padding: $s-size $m-size;
  border: 1px solid darken($off-white, 7%);
  color: $gray;
  background: $off-white;
}

.list-body {
  margin: 0 0 $m-size;

  @media (min-width: $desktop-breakpoint) {
    margin: 0 0 $l-size;
  }
}

.list-item {
  display: flex;
  flex-direction: column;
  padding: $s-size;
  border: 1px solid darken($off-white, 7%);
  border-top: none;
  text-decoration: none;
  color: $dark-gray;
  transition: background .3s ease;

  &:hover {
    background: $off-white;
  }

  @media (min-width: $desktop-breakpoint) {
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    padding: $m-size;
  }
}

.list-item--message {
  justify-content: center;
  align-items: center;
  padding: $m-size;
  color: $gray;

  &:hover {
    background: none;
  }
}

.list-item__title {
  margin: 0;
  word-break: break-all;
}

.list-item__subtitle {
  font-size: $fond-size-small;
  color: $gray;
}

.list-item__data {
  margin: $s-size 0 0;

  @media (min-width: $desktop-breakpoint) {
    margin: 0;
    padding: 0 0 0 $s-size;
  }
}
