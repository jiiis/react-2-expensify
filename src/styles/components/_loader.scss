.loader {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100vh;
  height: 100vh;
}

.loader__image {
  width: 6rem;
  height: 6rem;
}
