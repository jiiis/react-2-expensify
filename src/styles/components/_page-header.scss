.page-header {
  margin: 0 0 $l-size;
  padding: $l-size 0;
  background: $off-white;
}

.page-header__actions {
  margin: $m-size 0 0;
}

.page-header__title {
  margin: 0;
  font-weight: 300;

  span {
    font-weight: 700;
  }
}
