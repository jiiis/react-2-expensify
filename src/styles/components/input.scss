.text-input {
  padding: $s-size;
  height: 50px;
  border: 1px solid #cacccd;
  font-size: $fond-size-large;
  font-weight: 300;
}

.select {
  @extend .text-input;
}

.textarea {
  @extend .text-input;

  height: 10rem;
}
