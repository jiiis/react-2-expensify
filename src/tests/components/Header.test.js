import React from 'react'
import { shallow } from 'enzyme'
// import toJSON from 'enzyme-to-json'
// import ReactShallowRenderer from 'react-test-renderer/shallow'

import { Header } from '../../components/Header'

// Using react-test-renderer.
// test('should render Header', () => {
//   const renderer = new ReactShallowRenderer()
//
//   renderer.render(<Header />)
//
//   expect(renderer.getRenderOutput()).toMatchSnapshot()
// })

test('should render Header', () => {
  const wrapper = shallow(<Header startLogout={() => {}} />)

  expect(wrapper.find('h1').text()).toBe('Expensify')
  // expect(toJSON(wrapper)).toMatchSnapshot()
  expect(wrapper).toMatchSnapshot()
})

test('should call startLogout on button click', () => {
  const startLogoutSpy = jest.fn()
  const wrapper = shallow(<Header startLogout={startLogoutSpy} />)

  wrapper.find('button').simulate('click')

  expect(startLogoutSpy).toHaveBeenCalled()
})
