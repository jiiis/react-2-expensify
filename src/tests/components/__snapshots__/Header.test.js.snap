// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/dashboard"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/create"
  >
    Create expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/edit"
  >
    Edit expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/help"
  >
    Help
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
